client/my-sites/email/email-management/home/utils.js

```diff
@@ -114,7 +115,7 @@ export function resolveEmailPlanStatus( domain, emailAccount, isLoadingEmails )
 
 	if ( hasGSuiteWithUs( domain ) ) {
 		// Check for pending TOS acceptance warnings at the account level
-		if ( hasGoogleAccountTOSWarning( emailAccount ) ) {
+		if ( isPendingGSuiteTOSAcceptance( domain ) || ( emailAccount && hasGoogleAccountTOSWarning( emailAccount ) ) ) {
```

Comment:
The `emailAccount &&` is redundant here IMO. `hasGoogleAccountTOSWarning` will return false even `emailAccount` is null?

Reply:
My take was that this way it's a bit more explicit that we don't even call the helper function if `emailAccount` isn't defined -- the logic is easier to understand without needing to understand how `hasGoogleAccountTOSWarning()` handles an empty argument.

Reply:
It does feel like duplicating logic, however I understand what you mean.